lib/connections/limiter.go

```diff
@@ -73,22 +146,55 @@ func (lim *limiter) CommitConfiguration(from, to config.Configuration) bool {
 
 	lim.limitsLAN.set(to.Options.LimitBandwidthInLan)
 
+	// Set limits for devices
+	for _, v := range to.Devices {
+		if v.DeviceID == lim.myID {
+			// This limiter was created for local device. Should skip this device
+			continue
+		}
+
+		readLimiter, _ := lim.deviceReadLimiters.Load(v.DeviceID)
```

Comment:
You should not throw away the ok, in case the value is not there.
Also, you are not locking here, so you might be accessing deviceReadLimiters as the map is being swapped out.

This is also a reason why rebuildMap should potentially do it without swapping the maps.